### PR TITLE
Little fixes in documentation

### PR DESCRIPTION
**Pull request by willyrgf**
Reading the documentation I just found a few typos and some broken links. 

About the broken links, I understood that they are from deprecated documentations and I imagine that it'll be rewritten in this `next` version. Is that right? And, I think that's very important for anyone studying the project. 
Is that a satisfying way to handle it?

### CODE REVIEW

**Comment by bobbinth** [comment]
@willyrgf thank you! these look pretty good. One thing I should say is that we'll probably need to re-write most of these README files before merging `next` into `main` as most of the documentation will move here: https://maticnetwork.github.io/miden/

**Comment by willyrgf** [comment]
Nice @bobbinth. Considering that, do you think any of theses changes make sense?

File: docs/src/user_docs/assembly/main.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ In this note we use the following terms and notations:
 
 - $p$ is the modulus of the VM's base field which is equal to $2^{64} - 2^{32} + 1$.
 - A *binary* value means a field element which is either $0$ or $1$.
-- Inequality comparisons are assumed to be performed on integer representations of field elements in the range $[0, p)$.
+- Inequality comparisons are assumed to be performed on integer representations of field elements in the range $[0, p]$.
```
Comment (bobbinth) [inline docs/src/user_docs/assembly/main.md:25]: This is actually correct. The ")" signifies that `p` is not included (i.e., the range is exclusive).

**Review by bobbinth — CHANGES_REQUESTED** [review]
I think it doesn't hurt to merge this one. There was one comment that I left inline though.

File: docs/src/user_docs/assembly/main.md (hunk 1 above)
Comment (willyrgf) [reply docs/src/user_docs/assembly/main.md]: Oh yeah, I understood now, fixed.

**Review by bobbinth — APPROVED** [review]
Looks good! Thank you.

**Pull request merged**